### ISSUE TITLE
Using custom components for cell editing

### ISSUE THREAD

**Issue by markmywords** [AllenFang/react-bootstrap-table#162]
Hi there,

I was wondering if the ability of using custom components for cell editing wouldn't be a nice feature to have. Use cases could be file upload fields or date pickers. Adding such editors to the core would probably be out of scope but the ability to define and pass custom editor components to react-bootstrap-table certainly would be nice.

I already forked the project and had a go at it myself but I am not (yet!) a react.js pro, so I am unsure if my strategy for implement is the way to go or not :)

**Comment by AllenFang** [comment]
@markmywords, thanks you this component. I thinks the data picker can be included to the core, but upload field is a little bad idea, it need a custom component for user to define their editing component. So I need sometime to think about that :) In addition, if you have any questions, feel free to ask

**Comment by markmywords** [comment]
Hi and thank you for your fast reply. Yeah, a file uploader might not be a use case for a lot of people and I just wanted to use it as an example. What I needed for example was a select field where I can store an object id as value and display the object name at the same time. For this I just added the following to Editor.js:

```
else if(editable.type === 'custom'){
    var CustomWidget = editable.widget;
          return(
                <CustomWidget {...attr} attr={attr} options={editable.options} defaultValue={defaultValue} />
            )
}
```

It works fine but I don't know if this a good solution in react.js. I posted this issue to see if there is interest and maybe someone else is looking for a way to do something similar :)

And thanks for providing this great library btw!

**Comment by AllenFang** [comment]
It's seems like good at my first seeing, maybe you can make a PR and I can help you to make this cool feature be better :)

**Comment by dannysindra** [comment]
+1 for the datepicker in editing + inserting new data.

**Comment by battarro** [comment]
Any file uploader on the horizon?

**Comment by AllenFang** [comment]
hi @battarro and @dannysindra, Hope I can support it in recently, but I'm busy on my work, so PR is welcome. I promise I'll keep these asking feature in my mind, but there're a lot of things to do so that any improvement will be slow currently. sorry guys ;(

**Comment by AllenFang** [comment]
Released on ```v2.4.0```. Check this [example](https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/cell-edit/custom-cell-edit-table.js)

**Comment by AllenFang** [comment]
```v2.4.1``` enable to use date picker on row insert and cell editing, check [this](https://github.com/AllenFang/react-bootstrap-table/blob/master/examples/js/advance/edit-type-table.js#L38)
